### ISSUE TITLE
When we click Browser Back button its trigger wrong pageInit function.

### ISSUE THREAD

**Issue by rkaartikeyan** [nolimits4web/Framework7#1393]
* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* Framework7 Version: 1.5.3
* Platform and Target: in Browser
* Live Link or JSFiddle/Codepen: -Nil-

### What you did
I have Website(for mobile) with 7 pages. When i click main menu to go another page, in my Browser URL is like this (http://localhost/index.php#!/about.php) or (http://localhost/index.php#!/services.php) and i have pageInit functions for each pages.

### Expected Behavior
The website loaded with index.php and my url is (http://localhost/index.php).
I click About Us (http://localhost/index.php#!/about.php)
then i click Services (http://localhost/index.php#!/services.php)
then i click Contact Us (http://localhost/index.php#!/contact.php)

every thing Working Fine, Now i am in Contact Us page, and if i click Browser Back Button, its showing the content(html) of services.php in browser. Now it should trigger page init function for services.php.

### Actual Behavior
But instead Trigger Services page's Page Init function, its trigger about page init function.

**Comment by ZanderBrown** [comment]
do they have unique `data-page` values

**Comment by rkaartikeyan** [comment]
Yes, they have unique data-page values. i have tried with this also.

Home -> Contact Us -> Services -> About Us, Now if i click Back, Services Page showing, but Contact Us page Init event is triggering.

**Comment by ZanderBrown** [comment]
Due to the behaviour of `page:init` it won't trigger when going back to Services but it running for Contact Us is a bit odd

Can you share some code?

**Comment by rkaartikeyan** [comment]
http://f7.hyperlinkinfotech.com/ I have uploaded for you with simple navigation. please have a look.

**index.php**
```
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no, minimal-ui">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <meta name="theme-color" content="#2196f3">

    <title>Framework 7 Test - Welcome</title>

    <link rel="stylesheet" href="asserts/css/framework7.material.css">
    <link rel="stylesheet" href="asserts/css/framework7.material.colors.css">

    <link rel="stylesheet" href="css/material-icons.css">
    <link rel="stylesheet" href="css/kitchen-sink.css">
    <style type="text/css">
        .page-content {padding:30px;}
    </style>
</head>
<body class="theme-blue">
    <div class="statusbar-overlay"></div>

    <div class="views">
        <div class="view view-main">
            <div class="pages">
                <div data-page="index" class="page">
                    <div class="page-content">
                        <a href="index.php" title="Home">Home</a> |
                        <a href="about-us.php" title="About Us">About Us</a> |
                        <a href="services.php" title="Services">Services</a> |
                        <a href="contact-us.php" title="Contact Us">Contact Us</a>

                        <br /><br />

                        <div class="main-content">
                            Index
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>

    <script type="text/javascript" src="js/jquery-3.1.0.min.js"></script>
    <script type="text/javascript" src="asserts/js/framework7.js"></script>

    <script type="text/javascript">
        var myApp = new Framework7({
            modalTitle: 'Test Navigation',
            material: true,
            animatePages:false,
            cache: false,
            pushState: true
        });

        var $$ = Dom7;

        var mainView = myApp.addView('.view-main', {});

        myApp.onPageInit('index', function(){
            console.log('Index Page Init Function Triggered');
        }).trigger();

        myApp.onPageInit('about-us', function(){
            console.log('About Us Page Init Function Triggered');
        })

        myApp.onPageInit('services', function(){
            console.log('Services Page Init Function Triggered');
        })

        myApp.onPageInit('contact-us', function(){
            console.log('Contact Us Page Init Function Triggered');
        })

    </script>
</body>
</html>
```


**about-us.php**
```
<div data-page="about-us" class="page">
    <div class="page-content">
        <a href="index.php" title="Home">Home</a> |
        <a href="about-us.php" title="About Us">About Us</a> |
        <a href="services.php" title="Services">Services</a> |
        <a href="contact-us.php" title="Contact Us">Contact Us</a>

        <br /><br />

        <div class="main-content">
            About Us
        </div>
    </div>
</div>
```

**services.php**
```
<div data-page="services" class="page">
    <div class="page-content">
        <a href="index.php" title="Home">Home</a> |
        <a href="about-us.php" title="About Us">About Us</a> |
        <a href="services.php" title="Services">Services</a> |
        <a href="contact-us.php" title="Contact Us">Contact Us</a>

        <br /><br />

        <div class="main-content">
            Services
        </div>
    </div>
</div>
```

**contact-us.php**
```
<div data-page="contact-us" class="page">
    <div class="page-content">
        <a href="index.php" title="Home">Home</a> |
        <a href="about-us.php" title="About Us">About Us</a> |
        <a href="services.php" title="Services">Services</a> |
        <a href="contact-us.php" title="Contact Us">Contact Us</a>

        <br /><br />

        <div class="main-content">
            Contact Us
        </div>
    </div>
</div>
```

**Comment by ZanderBrown** [comment]
Ah i see know, F7 only keeps the current and previous page. So if the navigation is `Home > About > Services > Contact` then Contact & Services are kept in DOM.

To avoid reapplying event listeners F7 doesn't trigger `page:init` everytime the page becomes visible however as going back means About is now the previous page so F7 fetches it and runs `page:init` on it ensuring it's ready should the user go back again

TL;DR: this is a feature not a bug

**Comment by rkaartikeyan** [comment]
I go like Home > About > Services > Contact, and on page init i am fetching the content via Ajax and load it.

So when i am in Contact and if i click browser back button, it should show the Service page(its working) but it should trigger the Services Page Init Function and load the content. Instead its trigger About Page and loading its Content.

How this is as Feature???

**Comment by rkaartikeyan** [comment]
@ZanderBrown After spending many hour i found one thing When i click Browser Back button `app.router.back` & `app.router._back` functions triggering twice.

And Browser's Forward Button its working perfect.

**Comment by xiefangzhenz** [comment]
I have this issue in my project,do you have solve it?

**Comment by rkaartikeyan** [comment]
My issue resolved with 

```
view: {
    preloadPreviousPage: false,
}
```

Thanks, Framework7 Team. it's after a Year 🌊
